play-json/src/main/scala/play/api/libs/json/JsLookup.scala

```diff
@@ -105,6 +105,19 @@ sealed trait JsLookupResult extends Any with JsReadable {
   }
   def get: JsValue = toOption.get
   def getOrElse(v: => JsValue): JsValue = toOption.getOrElse(v)
+  def isEmpty: Boolean = this match {
+    case JsUndefined() => true
+    case JsDefined(_) => false
+  }
+  def isDefined: Boolean = !isEmpty
+
+  /**
+    * If this result is defined return `this`. Otherwise return `alternative`.
+    */
+  def orElse(alternative: => JsLookupResult): JsLookupResult = this match {
```

Comment:
`if (isDefined) this else alternative` (so we can keep the matching logic in one place)?